lib/app/sequence_base.rb

```diff
@@ -524,7 +524,8 @@ def validate_search_reply(klass, reply, search_params)
       def validate_read_reply(resource, klass)
         class_name = klass.name.demodulize
         assert !resource.nil?, "No #{class_name} resources available from search."
-        if resource.is_a? FHIR::DSTU2::Reference
+        resource_is_reference = resource.is_a? FHIR::DSTU2::Reference
+        if resource_is_reference
           read_response = resource.read
         else
           id = resource.try(:id)
```

Comment:
Might as well change this `resource&.id` maybe

Reply:
Changed.